homeassistant/components/vallox/__init__.py

```diff
@@ -114,16 +121,37 @@ def get_metric(self, metric_key: str) -> StateType:
 
         return value
 
+    def get_model(self) -> str | None:
+        """Return the model, if any."""
+        model = _api_get_model(self.metric_cache)
+
+        if not isinstance(model, str):
+            return None
+
+        if model == "Unknown":
+            return None
```

Comment:
```suggestion
        if not isinstance(model, str) or model == "Unknown":
            return None
```

Reply:
What are situations where model is not an instance of string? How and when can this happen?

Reply:
It can’t, but mypy seems to not take the 3rd party library type hints into account, so I needed to add this to silence it.

Reply:
In that case you should not check for `isinstance`, but use `cast()`. Or even better, adhere to PEP0484 in your package. (https://www.python.org/dev/peps/pep-0484/).

I faced the same issue and in my case adding a `py.typed` file to my package source did the trick.